hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/properties/TracingProperties.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.web3.evm.properties;
+
+import java.util.HashSet;
+import java.util.Set;
+import lombok.Data;
+import lombok.NonNull;
+import org.hyperledger.besu.datatypes.Address;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
+
+@Data
+@Validated
+@ConfigurationProperties(prefix = "hedera.mirror.web3.evm.tracing")
+public class TracingProperties {
+
+    private boolean enabled = false;
+
+    @NonNull
+    private Set<Address> contract = new HashSet<>();
```

Comment:
Have you tested this at runtime by populating the address in an `application.yml`? I don't think this will work without some custom converter. Instead, model as a String that supports and validates hex. Please test the below works as you would expect.

```suggestion
    private Set<@Hex(minLength = ADDRESS_LENGTH, maxLength = ADDRESS_LENGTH) String> contract = new HashSet<>();
```

Reply:
It is [valid](https://docs.jboss.org/hibernate/stable/validator/reference/en-US/html_single/#container-element-constraints) for use there. You need to add `ElementType.TYPE_USE` to the annotation.